data/data/baremetal/masters/main.tf

```diff
@@ -0,0 +1,115 @@
+# FIXME: This Terraform HCL file defines the 3 master nodes.  It uses the original ironic_nodes.json format,
+# flattened because Terraform v0.11 does not support nested data structures.  Maps may only be key/value
+# pairs.  We could use terraform's resource `count` provider to have just one resource declaration, but
+# the data would have to be structured differently.
+
+resource "ironic_node_v1" "openshift-master-0" {
+  name = "${var.nodes["nodes_0_name"]}"
+
+  target_provision_state = "active"
+  user_data              = "${var.ignition}"
+
+  ports = [
+    {
+      address     = "${var.nodes["nodes_0_ports_0_address"]}"
+      pxe_enabled = "${var.nodes["nodes_0_ports_0_pxe_enabled"]}"
+    },
+  ]
+
+  properties {
+    local_gb = "${var.nodes["nodes_0_properties_local_gb"]}"
+    cpu_arch = "${var.nodes["nodes_0_properties_cpu_arch"]}"
+  }
+
+  instance_info = {
+    image_source   = "${var.image_source}"
+    image_checksum = "${var.image_checksum}"
+    root_gb        = "${var.root_gb}"
+    root_device    = "${var.root_disk}"
+  }
+
+  driver = "ipmi"
+
+  driver_info {
+    ipmi_port      = "${var.nodes["nodes_0_driver_info_ipmi_port"]}"
```

Comment:
I think this is a concern for the bare metal testers who didn't want to specify a port.  This whole section is also hard-coded to IPMI.  It might be a good idea for me to see if I can't extract driver_info into it's own map so users can specify whatever they want - ipmi, idrac, etc.

Reply:
The code in https://github.com/metalkube/baremetal-operator/blob/master/pkg/bmc/access.go (as well as idrac.go and ipmi.go in that directory) may be useful, at least as a reference. Notice that the driver_info *keys* change based on the driver.

Reply:
Yes, that's helpful - thanks. 

The main problem I have is terraform, it doesn't have very robust data structures.  A list can only be strings.  A map can only be string/string pairs.  I have to think about how a user is going to represent the hardware info in installer-config.yaml for kni-installer.  The easiest way is something like this

```
driver_info_master_0:
  drac_username: admin
  drac_password: potato
  drac_address: ...
driver_info_master_1:
  drac_username: admin
  drac_password: potato
  drac_address: ...
```

But then you have to know in advance how masters there are for the terraform template.  It's very limiting.

Reply:
I would stay away from "drac" as a prefix since as soon as we have some other hardware support in place it will look odd. BMC would work. The address field can be a URL like idrac+https://host:port/path so we get the type from that.

The username and passwords will end up going into a secret (one per host) so using separate fields will save you having to parse them out of the URL yourself.

So, the format you have looks OK to me.